### PR TITLE
Added sync for incubator

### PR DESCRIPTION
**Pull request by Coding-Hen**
TODO: Make server aware of entity currently throws entity unknown to the server error when sending across the metadata

### CODE REVIEW

File: NitroxClient/GameLogic/Spawning/Metadata/IncubatorMetadataProcessor.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    {
+        public override void ProcessMetadata(GameObject gameObject, IncubatorMetadata metadata)
+        {
+            Incubator incubator = gameObject.GetComponent<Incubator>();
+            incubator.powered = metadata.Powered;
+            incubator.hatched = metadata.Hatched;
+
+            bool terminalActive = (bool)incubator.ReflectionGet("terminalActive");
+            incubator.computerTerminal.SetActive(terminalActive);
+            incubator.ReflectionSet("emissiveIntensity", (float)(terminalActive ? 1 : 0));
```
Comment (Dessix) [inline NitroxClient/GameLogic/Spawning/Metadata/IncubatorMetadataProcessor.cs:17]:
```suggestion
            incubator.ReflectionSet("emissiveIntensity", (terminalActive ? 1f : 0f));
```

File: NitroxClient/GameLogic/Spawning/Metadata/IncubatorMetadataProcessor.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+{
+    public class IncubatorMetadataProcessor : GenericEntityMetadataProcessor<IncubatorMetadata>
+    {
+        public override void ProcessMetadata(GameObject gameObject, IncubatorMetadata metadata)
+        {
+            Incubator incubator = gameObject.GetComponent<Incubator>();
+            incubator.powered = metadata.Powered;
+            incubator.hatched = metadata.Hatched;
+
+            bool terminalActive = (bool)incubator.ReflectionGet("terminalActive");
```
Comment (Dessix) [inline NitroxClient/GameLogic/Spawning/Metadata/IncubatorMetadataProcessor.cs:15]:
```suggestion
            var terminalActive = incubator.ReflectionGet<bool>("terminalActive");
```

File: NitroxClient/GameLogic/Spawning/Metadata/IncubatorMetadataProcessor.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+            incubator.computerTerminal.SetActive(terminalActive);
+            incubator.ReflectionSet("emissiveIntensity", (float)(terminalActive ? 1 : 0));
+            incubator.ReflectionCall("UpdateMaterialsEmissive");
+            incubator.terminalLight.intensity = (float)incubator.ReflectionGet("emissiveIntensity") * 1.85f;
```
Comment (Dessix) [inline NitroxClient/GameLogic/Spawning/Metadata/IncubatorMetadataProcessor.cs:19]:
```suggestion
            incubator.terminalLight.intensity = incubator.ReflectionGet<float>("emissiveIntensity") * 1.85f;
```

File: NitroxClient/GameLogic/Spawning/Metadata/IncubatorMetadataProcessor.cs (hunk 2 above)
Comment (Coding-Hen) [reply NitroxClient/GameLogic/Spawning/Metadata/IncubatorMetadataProcessor.cs:15]: Unless I am mistaken I do not believe this works due to how the ReflectionHelper returns the object type?

Comment (Dessix) [reply NitroxClient/GameLogic/Spawning/Metadata/IncubatorMetadataProcessor.cs:15]: My apologies, retracting this one and the other; I misinterpreted the intent of T in the `ReflectionHelper`s.

```
public static object ReflectionGet<T>(this T o, string fieldName, bool isPublic = false, bool isStatic = false)
//                  ^ expected T
```

Comment (Coding-Hen) [reply NitroxClient/GameLogic/Spawning/Metadata/IncubatorMetadataProcessor.cs:15]: Would be cool if we could do this though, I would have thought even with T though would it not have issues with say in this case passing in bool but the class being passed in as Incubator would that have an issue with the return type?

**Comment by Sunrunner37** [comment]
Looks like this is still in draft.  What is remaining on this?

**Comment by Coding-Hen** [comment]
Fixed with 7a949b7214d6df20838e8e28a60a373b36450a30

**Pull request closed without merging**